### PR TITLE
[CARBONDATA-99]Complex type column filters with like and not like not working

### PR DESCRIPTION
**Pull request by sujith71955**
For complex type filter queries if query contains filter expression rather than BinaryExpression the system was not able to get the dimensions which are involved in the particular filter expression for executing complex type filter column expressions.

### CODE REVIEW

**Comment by sujith71955** [comment]
retest this please

File: core/src/main/java/org/carbondata/query/carbon/executor/util/QueryUtil.java [hunk 1]
```diff
@@ -1108,6 +1108,8 @@ private static int fillChildrenDetails(int[] eachComplexColumnValueSize,
     if (filterExpression instanceof BinaryLogicalExpression) {
       BinaryLogicalExpression logicalExpression = (BinaryLogicalExpression) filterExpression;
       dimensionResolvedInfos.addAll(logicalExpression.getColumnList());
+    } else if(null!=filterExpression) {
```
Comment (kumarvishal09) [inline core/src/main/java/org/carbondata/query/carbon/executor/util/QueryUtil.java]: Why this null check is required? Filter expression cannot be null

**Comment by kumarvishal09** [comment]
Please resolve the merge conflict

File: integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/complexType/TestComplexTypeQuery.scala [hunk 2]
```diff
@@ -41,6 +41,17 @@ class TestComplexTypeQuery extends QueryTest with BeforeAndAfterAll {
...
      sql("create table complexhivetable(deviceInformationId int, channelsId string, ROMSize string, ROMName String, purchasedate string, mobile struct<imei:string, imsi:string>, MAC array<string>, locationinfo array<struct<ActiveAreaId:int, ActiveCountry:string, ActiveProvince:string, Activecity:string, ActiveDistrict:string, ActiveStreet:string>>, proddate struct<productionDate:string,activeDeactivedate:array<string>>, gamePointId double,contractNumber double)row format delimited fields terminated by ',' collection items terminated by '$' map keys terminated by ':'")
      sql("LOAD DATA local inpath './src/test/resources/complextypesample.csv' INTO table complexhivetable");
+     sql("create table complex_filter(test1 int, test2 array<String>,test3 array<bigint>,test4 array<int>,test5 array<decimal>,test6 array<timestamp>,test7 array<double>) STORED BY 'org.apache.carbondata.format'")
```
Comment (nareshpr) [inline integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/complexType/TestComplexTypeQuery.scala]: Add drop table for complex_filter. Add test cases for array column like filter. Is it possible to use existing table for filter queries or is it specific scenario?

Comment (sujith71955) [inline integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/complexType/TestComplexTypeQuery.scala]: added as per comments,, regarding table there is scenario of BigInt,BigDecimals

**Comment by kumarvishal09** [comment]
@sujith71955  Please raise jira issue and fix null check issue

**Comment by sujith71955** [comment]
JIRA ID : CARBONDATA-99

**Comment by kumarvishal09** [comment]
retest this please

**Comment by kumarvishal09** [comment]
LGTM

**Comment by sujith71955** [comment]
retest this please

**Pull request merged**